src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets

```diff
@@ -396,12 +396,17 @@ Copyright (c) .NET Foundation. All rights reserved.
     -->
   <Target Name="IncludeTargetingPackReference" BeforeTargets="_CheckForInvalidConfigurationAndPlatform;CollectPackageReferences;CheckForImplicitPackageReferenceOverrides"
           Condition="'$(TargetFrameworkMoniker)' != '' and '$(TargetFrameworkIdentifier)' == '.NETFramework' and '$(AutomaticallyUseReferenceAssemblyPackages)' == 'true'">
+    
+    <!-- Skip this task if is using static graph, because in that case we don't want the NuGet packages referenced to change based on machine state, so we should always
+         reference the reference assembly package. -->
     <GetReferenceAssemblyPaths
         TargetFrameworkMoniker="$(TargetFrameworkMoniker)"
         RootPath="$(TargetFrameworkRootPath)"
         TargetFrameworkFallbackSearchPaths="$(TargetFrameworkFallbackSearchPaths)"
         BypassFrameworkInstallChecks="true"
-        SuppressNotFoundError="true">
+        SuppressNotFoundError="true"
+        Condition="'$(IsGraphBuild)' != 'true'"
```

Comment:
Isn't this hiding the real problem that for some reason static graph based restore leads to different reference assembly paths?

Reply:
Not as far as I understand it.  I think the different reference assembly paths were caused because the reference assemblies installed in Program Files would be used if they were installed, otherwise the NuGet packages with those reference assemblies would be referenced and they would be used from the package.

This change makes it so that for graph build, it will always use the reference assembly NuGet packages, whether or not the reference assemblies are installed in Program Files.

Reply:
It could be that I also misunderstand the problem. Does the static graph based restore cause any difference in the ref assembly paths? If yes, then the root cause for that should be a bug right? Because using static graph should not lead to observable differences in build output.